terra/compute/base.py

```diff
@@ -300,13 +302,39 @@ def configure_logger(sender, **kwargs):
         sender._log_file = os.devnull
       if settings.processing_dir:
         os.makedirs(settings.processing_dir, exist_ok=True)
+
+      if settings.logging.server.family == "AF_UNIX":
+        # import pdb; pdb.set_trace()
+        socket_dir = Path(settings.logging.server.listen_address).parent
+        os.makedirs(socket_dir, exist_ok=True)
+        temp_filename = socket_dir / (
+            ".terra_test_" + settings.terra.uuid + ".sock")
+        temp_sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+        try:
+          temp_sock.bind(str(temp_filename))
+          temp_sock.close()
+        except OSError:
+          # /dev/shm should always work on Linux
+          settings.logging.server.listen_address = str(
+              Path("/dev/shm") / (
+                  ".terra_log_" + settings.terra.uuid + ".sock"))
+          logger.info(
+              f"{socket_dir} cannot handle file sockets. "
+              f"Using {settings.logging.server.listen_address} instead.")
```

Comment:
Would it make more sense to throw an informative error rather than revise the setting provided by the user?

Reply:
I would rather have it "just work" than have to go back and change the settings files. I think the average user doesn't want to have to remember "Oh, on this file system I need to manually specify the listen address." I know the lock dir and sql dir do _not_ work this way yet, but one PR step at a time.